Library/Homebrew/formula.rb

```diff
@@ -405,11 +413,19 @@ def linked_keg
     Pathname.new("#{HOMEBREW_LIBRARY}/LinkedKegs/#{name}")
   end
 
+  def latest_head_prefix
+    head_prefixes = installed_prefixes.select do |pn|
+      pn.basename.to_s.start_with?("HEAD")
```

Comment:
I finally got it. Yes, it seems to work. Please, correct me if I'm wrong: `head_prefix_up_to_date?(prefix)` for this logic is called in two cases: 

1. When it definitely returns false (revision haven't been skipped since that prefix had been installed, which implies that stable haven't been changed) 
2. When we skip revision to `0` and it increments until it's the same as `prefix` has, when they are the same, `head_prefix_up_to_date` can be called, but it'll return false.